openedx_events/event_bus/avro/tests/test_avro.py

```diff
@@ -78,17 +95,80 @@ def test_all_events(self):
         for signal in OpenEdxPublicSignal.all_events():
             if signal.event_type in KNOWN_UNSERIALIZABLE_SIGNALS:
                 continue
-            test_data = {}
+            test_data = generate_test_data_for_signal(signal)
             serializer = AvroSignalSerializer(signal)
-            for key, curr_class in signal.init_data.items():
-                example_data = generate_test_event_data_for_data_type(curr_class)
-                example_data_processed = curr_class(**example_data)
-                test_data.update({key: example_data_processed})
             serialized = serializer.to_dict(test_data)
             deserializer = AvroSignalDeserializer(signal)
             deserialized = deserializer.from_dict(serialized)
             self.assertDictEqual(deserialized, test_data)
 
+    def test_evolution_is_forward_compatible(self):
+        """
+        Test current version of events is forward compatible with stored schemas.
+
+        There's no assert because as long as the test doesn't raise an error, the new schema is
+        forward-compatible with the original.
+        """
+        for signal in OpenEdxPublicSignal.all_events():
+            if signal.event_type in KNOWN_UNSERIALIZABLE_SIGNALS:
+                continue
+            test_data = generate_test_data_for_signal(signal)
+            serializer = AvroSignalSerializer(signal)
+            schema_dict = serializer.schema
+
+            # write to bytes using current schema
+            current_out = io.BytesIO()
+            data_dict = serializer.to_dict(test_data)
+            schemaless_writer(current_out, schema_dict, data_dict)
+            current_out.seek(0)
+            current_event_bytes = current_out.read()
+
+            # get stored schema
+            stored_schema = load_schema(f"{os.path.dirname(os.path.abspath(__file__))}/schemas/"
```

Comment:
We don't have a full process for updating schemas so I'm hesitant to make much documentation about it. In any case, if there's no file, this would be an issue with how schemas are added as opposed to updated. I guess we can document that somewhere.

Reply:
Could you even just add a comment above this line that if the schema is missing, they probably need to run the management command to generate a file for a new event?